flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/Mail.java

```diff
@@ -71,8 +77,13 @@ public Mail(
         this.actionExecutor = actionExecutor;
     }
 
+    public MailboxExecutor.MailOptions getMailOptions() {
+        return mailOptions;
+    }
+
     public int getPriority() {
-        return priority;
+        /** See {@link MailboxExecutor.MailOptions#setInterruptible()}. */
+        return mailOptions.isInterruptible() ? TaskMailbox.MIN_PRIORITY : priority;
```

Comment:
I see the `MailboxProcessor#getMainMailboxExecutor` create the `MailboxExecutorImpl` with `MIN_PRIORITY`, which means for this executor `#execute` will put mail with `MIN_PRIORITY`. Should this also be changed?

Reply:
That should be fine AFAIK. `MIN_PRIORITY` is used for the top level mails, that no one should ever `yield` to. `yield()`. Operators, which are the only ones who should ever call `.yield()`,  have priorities always `> MIN_PRIORITY`.

I dunno, I think `priority` name here might be a bit misleading. It's more like ordering id. Entity with `orderingIdA` that, is not allowed to yield to an entity with `orderingIdB`, where `orderingIdA > orderingIdB`. This way of yielding, we were calling `yield-to-downstream` - operators were allowed to only yield to a downstream entity, to relief back-pressure.

Reply:
> That should be fine AFAIK. `MIN_PRIORITY` is used for the top level mails, that no one should ever `yield` to. `yield()`. Operators, which are the only ones who should ever call `.yield()`, have priorities always `> MIN_PRIORITY`.

I misread that part. This should be fine.